skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java

```diff
@@ -137,27 +138,31 @@ public void doAction() throws Exception {
         }
     }
     
-    public void stopSession(String userID, String type, String sessionID) throws Exception {
+    public void deleteSession(String userID, String type, String sessionID) throws Exception {
         // kill the session specified by sessionID
         log.debug("Stopping " + type + " session: " + sessionID);
         String k8sNamespace = K8SUtil.getWorkloadNamespace();
         
         String podName = K8SUtil.getJobName(sessionID, type, userID);
-        String[] cmd = new String[] {
-            "kubectl", "delete", "--namespace", k8sNamespace, "job", podName};
-        execute(cmd);
+        delete(k8sNamespace, "job", podName);
         
         if (!SESSION_TYPE_HEADLESS.equals(type)) {
             String ingressName = K8SUtil.getIngressName(sessionID, type);
-            cmd = new String[] {
-                "kubectl", "delete", "--namespace", k8sNamespace, "ingressroute", ingressName};
-            execute(cmd);
+            delete(k8sNamespace, "ingressroute", ingressName);
             
             String serviceName = K8SUtil.getServiceName(sessionID, type);
-            cmd = new String[] {
-                "kubectl", "delete", "--namespace", k8sNamespace, "service", serviceName};
+            delete(k8sNamespace, "service", serviceName);
+        }
+    }
+    
+    private void delete(String k8sNamespace, String type, String name) throws InterruptedException, IOException {
+        try {
+            String[] cmd = new String[] {
+                "kubectl", "delete", "--namespace", k8sNamespace, type, name};
             execute(cmd);
+        } catch (IOException ex) {
```

Comment:
You may as well catch 'Exception' in stead of just 'IOException' to make sure we get catch the runtime ones as well.

Reply:
Sure. This changes the initial intent of the ticket, but works.
Done.